### PR TITLE
get allways fresh device type, add it also for app resource

### PR DESCRIPTION
**Pull request by ahilles107**
At moment when we initialize DeviceDetection userAgent is not populated yet. With this change we will have allwys correct current_device value.

### CODE REVIEW

**Comment by ahilles107** [comment]
it fix #103

**Comment by ahilles107** [comment]
hhvm fails on tests with ```No code coverage driver available``` error message.

File: Locator/FileLocator.php [hunk 1]
```diff
@@ -251,7 +246,8 @@ protected function locateAppResource($name, $dir = null, $first = true)
         $parameters = array(
             '%app_path%'      => $this->path,
             '%current_theme%' => $this->lastTheme,
-            '%template%'      => substr($name, strlen('views/')),
+            '%current_device%' => $this->activeTheme->getDeviceType(),
+            '%template%'      => substr($name, strlen('views/'))
```
Comment (lsmith77) [inline Locator/FileLocator.php]: please keep the trailing ``,`` here

**Comment by lsmith77** [comment]
can you also have a look if this relates to what we did in https://github.com/liip/LiipThemeBundle/pull/113 ?

**Comment by ahilles107** [comment]
even calling setCurrentTheme in contructor don't fix that. Looks like somehow ```setUserAgent``` is called many times without user agent and we can't use it in ```FileLocator``` constructor. This PR fixes it but for now, i will try to check where is real big but it can take time. Merging is up to You. In our project we can use fork.

**Comment by lsmith77** [comment]
can you rebase to get the HHVM issue fixed

**Comment by lsmith77** [comment]
aside from this it would be good if you could add a test case

**Comment by ahilles107** [comment]
i will do it

**Comment by ahilles107** [comment]
@lsmith77 if tests are ok i will rebase it. Let me know.

**Comment by lsmith77** [comment]
i am still a bit nervous if we are not going to break something again .. but in the previous changes I didn't insist on a test .. so hopefully if we do break something again, then we have a tewst case to prevent another round of regressions

**Comment by ahilles107** [comment]
Like I said - we will look for real issue soon. Expect PR.

**Pull request merged**